packages/react/src/components/CodeEditor/CodeEditor.jsx

```diff
@@ -0,0 +1,216 @@
+import Editor from '@monaco-editor/react';
+import { CodeSnippetSkeleton, CopyButton, Button } from 'carbon-components-react';
+import PropTypes from 'prop-types';
+import { Upload16 } from '@carbon/icons-react';
+import classnames from 'classnames';
+
+import { settings } from '../../constants/Settings';
+
+const { prefix: carbonPrefix, iotPrefix } = settings;
+
+export const updateEditorAttribute = (disabled, editorValue) => {
+  const textarea = document.getElementsByClassName('inputarea monaco-mouse-cursor-text')[0];
+
+  if (disabled && !!editorValue.current) {
+    textarea.setAttribute('disabled', '');
+  } else if (textarea?.hasAttribute('disabled')) {
+    textarea.removeAttribute('disabled');
+  }
+};
+
+const propTypes = {
+  /** Initial value for the editor */
+  initialValue: PropTypes.string,
+  /** the language being written in the editor */
+  language: PropTypes.string,
+  /** Callback called when editor copy icon is pressed */
+  onCopy: PropTypes.func,
+  /** List of types that the upload input accept */
+  accept: PropTypes.arrayOf(PropTypes.string),
+  /** Light theme */
+  light: PropTypes.bool,
+  /** Boolean to define if upload button should be render or not */
+  hasUpload: PropTypes.bool,
+  /** Callback on code editor change */
+  onCodeEditorChange: PropTypes.func,
+  /** Boolean to disabled code editor */
+  disabled: PropTypes.bool,
+  /** All the labels that need translation */
+  i18n: PropTypes.shape({
+    copyBtnDescription: PropTypes.string,
+    copyBtnFeedBack: PropTypes.string,
+    uploadBtnDescription: PropTypes.string,
+  }),
+  testId: PropTypes.string,
+};
+
+const defaultProps = {
+  initialValue: null,
+  language: 'css',
+  onCopy: null,
+  accept: ['.css'],
+  light: false,
+  hasUpload: false,
+  onCodeEditorChange: null,
+  disabled: false,
+  i18n: {
+    copyBtnDescription: 'Copy content',
+    copyBtnFeedBack: 'Copied',
+    uploadBtnDescription: 'Upload your file',
+  },
+  testId: 'code-editor',
+};
+const CodeEditor = ({
+  initialValue,
+  language,
+  onCopy,
+  accept,
+  hasUpload,
+  onCodeEditorChange,
+  light,
+  disabled,
+  i18n,
+  testId,
+}) => {
+  const mergedI18n = useMemo(() => ({ ...defaultProps.i18n, ...i18n }), [i18n]);
+
+  const editorValue = useRef();
+  const inputNode = useRef(null);
+
+  const [codeEditorValue, setCodeEditorValue] = useState(initialValue);
+
+  useEffect(() => {
+    updateEditorAttribute(disabled, editorValue);
+
+    const button = document.getElementsByClassName('iot--code-editor-copy')[0];
+
+    if (disabled) {
+      button.setAttribute('disabled', '');
+    } else if (button?.hasAttribute('disabled')) {
+      button.removeAttribute('disabled');
+    }
+  }, [disabled]);
+
+  /**
+   *
+   * @param {func} _editorValue - a method that returns the current value of the editor
+   * @param {object} val - instance of the monaco editor
+   */
+  const handleEditorDidMount = (_editorValue, val) => {
```

Comment:
In CardCodeEditor you have the following code. 
  ```
  /**
   *
   * @param {object} editor - instance of the editor
   * @param {object} _monaco - instance of monaco
   */
  // eslint-disable-next-line no-unused-vars
  const handleEditorDidMount = (editor, _monaco) => {
    editorValue.current = editor;
  };
  
  ```
  
  where you are setting the first parameter as the current value, whereas you are setting 2nd parameter in here. 
  
  Is this correct? I have just done a visual code review I'm simply observing this. I may be completely wrong here!

Reply:
Good catch, updated the `CodeEditor`, to be like `CardCodeEditor `... the function return has changed to updating it to both be the same ;)